presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -257,4 +259,21 @@ public Session getSession()
             return session;
         }
     }
+
+    private static class Result
+            implements Rule.Result
+    {
+        private Optional<PlanNode> transformed = Optional.empty();
+
+        @Override
+        public Rule.Result transformTo(PlanNode planNode)
+        {
+            return this;
+        }
+
+        private Optional<PlanNode> getTransformed()
```

Comment:
I guess the idea is that you could extract such static class from nesting class and it "should" work without access modifications. This of it, if you cannot access private methods of super class then why should you manipulate with private fields/methods of nested static class.
I know that nesting class can access anything from nested class, but I prefer that there is some convention (e.g: how such class would look like if it wasn't nested).